cpp/include/raft/matrix/batched_rearrange.cuh

```diff
@@ -15,73 +15,78 @@
  */
 #pragma once
 
-#include <raft/core/error.hpp>
+#include <raft/core/device_mdspan.hpp>
 #include <raft/core/device_resources.hpp>
+#include <raft/core/error.hpp>
 #include <raft/matrix/detail/batched_rearrange.cuh>
 
 namespace raft {
 namespace matrix {
 
 /**
- * In-place gather elements from a row-major source matrix into a destination range according to a map. The length of the map is equal to the number of rows. Batching is done on columns and an additional scratch space of shape n_rows * cols_batch_size is created. For each batch, chunks of columns from each row are copied into the appropriate location in the scratch space and copied back to the corresponding locations in the input matrix.  
+ * @brief In-place gather elements in a row-major matrix according to a
+ * map. The length of the map is equal to the number of rows.
+ * Batching is done on columns and an additional scratch space of
+ * shape n_rows * cols_batch_size is created. For each batch, chunks
+ * of columns from each row are copied into the appropriate location
+ * in the scratch space and copied back to the corresponding locations
+ * in the input matrix
+ *
  * @tparam value_idx
  * @tparam value_t
+ *
  * @param[in] handle raft handle
- * @param[out] in input matrix (n_rows * n_cols)
+ * @param[inout] in input matrix (n_rows * n_cols)
  * @param[in] map map containing the order in which rows are to be rearranged (n_rows)
- * @param  D Number of columns of the input/output matrices
- * @param  N Number of rows of the input matrix
- * @param col_batch_size column batch size
+ * @param[in] col_batch_size column batch size
  */
 template <typename T, typename IdxT>
 void batched_gather(raft::device_resources const& handle,
```

Comment:
Could we rename this function to `gather_inplace`? That explain what it is doing. The 'batching` is an implementation detail imho.

Similarly, we should change the filename and `batched_scatter`.